clients/java/src/main/java/io/camunda/zeebe/client/impl/command/DeleteTenantCommandImpl.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.zeebe.client.impl.command;
+
+import io.camunda.zeebe.client.api.ZeebeFuture;
+import io.camunda.zeebe.client.api.command.DeleteTenantCommandStep1;
+import io.camunda.zeebe.client.api.command.FinalCommandStep;
+import io.camunda.zeebe.client.api.response.DeleteTenantResponse;
+import io.camunda.zeebe.client.impl.http.HttpClient;
+import io.camunda.zeebe.client.impl.http.HttpZeebeFuture;
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+import org.apache.hc.client5.http.config.RequestConfig;
+
+public final class DeleteTenantCommandImpl implements DeleteTenantCommandStep1 {
+  private long tenantKey;
+  private final HttpClient httpClient;
+  private final RequestConfig.Builder httpRequestConfig;
+
+  public DeleteTenantCommandImpl(final HttpClient httpClient) {
+    this.httpClient = httpClient;
+    httpRequestConfig = httpClient.newRequestConfig();
+  }
+
+  @Override
+  public DeleteTenantCommandStep1 tenantKey(final long tenantKey) {
+    this.tenantKey = tenantKey;
+    return this;
+  }
+
+  @Override
+  public FinalCommandStep<DeleteTenantResponse> requestTimeout(final Duration requestTimeout) {
+    httpRequestConfig.setResponseTimeout(requestTimeout.toMillis(), TimeUnit.MILLISECONDS);
+    return this;
+  }
+
+  @Override
+  public ZeebeFuture<DeleteTenantResponse> send() {
+    ArgumentUtil.ensureGreaterThan("tenantKey", tenantKey, 0);
```

Comment:
Do we need this check? We don't usually seem to check this the key and this seems like the wrong layer to assume anything about the key encoding.

Reply:
Checking for known invalid values in the client makes sense for me. The key is an opaque value though: we only know that it can't be negative today because we know it is generated, but there's no guarantee that will always be the case. 
In any case, the API will [switch to String keys](https://camunda.slack.com/archives/C06UKS51QV9/p1732305476336959?thread_ts=1731429420.606829&cid=C06UKS51QV9) in the near future (for unrelated reasons), but that will make it clearer that keys are opaque values and clients shouldn't parse or validate them.